File: src/main/java/dev/wnuke/NukeClient/module/modules/FullBright.java
package dev.wnuke.NukeClient.module.modules;

import dev.wnuke.NukeClient.module.Category;
import dev.wnuke.NukeClient.module.Module;

public class FullBright extends Module {
    public FullBright() {
        super("Full Bright", Category.RENDER, "Makes everything maximum brightness");
    }

    private double originalBrightness = 1.0;

    @Override
    public void onEnabled() {
        originalBrightness = mc.gameSettings.gamma;
    }

    public void onTick() {
        if (mc.gameSettings.gamma < 16.0) mc.gameSettings.gamma += 1.2;
    }

    @Override
    public void onDisabled() {
        mc.gameSettings.gamma = originalBrightness;
    }
}

File: src/main/kotlin/dev/wnuke/nukeclient/module/modules/FullBright.kt
package dev.wnuke.nukeclient.module.modules

import dev.wnuke.nukeclient.event.Subscribe
import dev.wnuke.nukeclient.event.events.Tick
import dev.wnuke.nukeclient.module.Category
import dev.wnuke.nukeclient.module.Module

class FullBright : Module("full-bright", "Let's you see in the dark.", Category.RENDER) {
    private var startGamma = 0.0

    override fun onEnable() {
        startGamma = mc.options.gamma
    }

    override fun onDisable() {
        mc.options.gamma = startGamma
    }

    @Subscribe
    fun onTick(event: Tick) {
        if (mc.options.gamma < 16.0) {
            mc.options.gamma = 16.0
        }
    }
}